apm-lambda-extension/app/run.go

```diff
@@ -201,9 +207,13 @@ func (app *App) processEvent(
 	// 2) [Backup 1] RuntimeDone is triggered upon reception of a Lambda log entry certifying the end of the execution of the current function
 	// 3) [Backup 2] If all else fails, the extension relies of the timeout of the Lambda function to interrupt itself 100 ms before the specified deadline.
 	// This time interval is large enough to attempt a last flush attempt (if SendStrategy == syncFlush) before the environment gets shut down.
+
 	select {
-	case <-app.apmClient.AgentDoneSignal:
-		app.logger.Debug("Received agent done signal")
+	case <-app.apmClient.WaitForFlush():
+		app.logger.Debug("APM client has pending flush signals")
+
+		// Reset flush state for future events.
+		app.apmClient.ResetFlush()
```

Comment:
Should we call this with a `defer` at the top of the function? In case the flushed request comes, but runtimeDone or the timer also fire.